Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+    $contentVersion = "variables('dataConnectorVersion$($templateKindByCounter[$TemplateCounter])')";
+    $contentTemplateName = "variables('dataConnectorTemplateName$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)')";
+    $contentId = "variables('_dataConnectorContentId$($templateKindByCounter[$TemplateCounter])$($global:connectorCounter)')";
+    $resoureKind = $templateContentTypeByCounter[$TemplateCounter];
+    if($resoureKind -eq "DataConnector")
+    {
+        $resoureKindTag = "dc";
+    }
+    else {
+        $resoureKindTag = "rdc";
+    }
+
+    $title = $ccpItem.title;
+    $displayName = $title;
+    $randomNumber = Get-Random
+    return [PSCustomObject]@{
+        type       = "Microsoft.OperationalInsights/workspaces/providers/contentTemplates";
+        apiVersion = $contentResourceDetails.metadataApiVersion; # "2023-04-01-preview";
+        name        = "[concat(parameters('workspace'),'/Microsoft.SecurityInsights/', $contentTemplateName, $randomNumber)]";
+        location   = "[parameters('workspace-location')]";
+        dependsOn  = @(
+            "$($contentResourceDetails.dependsOn)"
+        );
+        properties = [PSCustomObject]@{
+            contentId  =  "[$contentId]";
+            displayName = $displayName;
+            contentKind = $templateContentTypeByCounter[$TemplateCounter];
+            mainTemplate = [PSCustomObject]@{
+                '$schema'      = "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#";
+                contentVersion = "[$contentVersion]";
+                parameters     = [PSCustomObject]@{};
+                variables      = [PSCustomObject]@{};
+                resources      = @(
+                )
+            };
+            packageKind = "Solution";
+            packageVersion = "[variables('_solutionVersion')]";
+            packageName = "[variables('_solutionName')]";
+            contentProductId = "[concat(take(variables('_solutionId'), 50),'-','$resoureKindTag','-', uniqueString(concat(variables('_solutionId'),'-','$resoureKind','-',$contentId,'-', $contentVersion)))]";
+            packageId = "[variables('_solutionId')]";
+            contentSchemaVersion = $contentResourceDetails.contentSchemaVersion;
+            version = "[variables('_solutionVersion')]";
+        }
+    }
+}
+
+function Get-ArmResource($name, $type, $kind, $properties){
+    [hashtable]$apiVersion = @{
+        "Microsoft.SecurityInsights/dataConnectors" = "2022-12-01-preview";
+        "Microsoft.SecurityInsights/dataConnectorDefinitions" = "2022-09-01-preview";
+        "Microsoft.OperationalInsights/workspaces/tables" = "2021-03-01-privatepreview";
+        "Microsoft.Insights/dataCollectionRules" = "2021-09-01-preview";
+    }
+
+    return [PSCustomObject]@{
+        name       = $name;
+        apiVersion = $apiVersion[$type]
+        type       = $type;
+        location   = "[parameters('workspace-location')]";
```

Comment:
check the location

Reply:
it is required else it fails 
![image](https://github.com/Azure/Azure-Sentinel/assets/107389644/0a15027b-b228-4db7-805f-bdfea7e80888)